### PR TITLE
Changed input to optocoupler and text regarding invert signal.

### PR DESCRIPTION
**Pull request by MatsA**
Changed circuit at optocoupler input making a lesser load. Optocoupler now connected between +12 VDC and ST1 signal. Corrected info about inverting signal.

### CODE REVIEW

**Comment by tkurki** [comment]
Please add a description and a descriptive title: what are you changing and especially *why*? PR titles form the next version changelog.

Also: I don't know much if anything about ST1, but the new picture seems strange to me: only 12 v and ground are connected to the optocoupler?

**Comment by MatsA** [comment]
Sorry ! OK, updated description and title and hope it's OK. In the former version the optocoupler input was connected between ST1 DATA and GOUND and now connected between ST1  +12VDC and DATA. It could look a bit strange since ST1 DATA is connected to the optocoupler **labeled** ground.

It should also be said that when the conversion was done to the new doc format the source for the conversion was not the latest, since the changed picture, above, was in the latest version..... Check here https://github.com/SignalK/signalk-server/blob/d2d9b27174924908bbf5ce1fc438004cfc3bfbac/Seatalk(GPIO).md
 /Mats

**Comment by KEGustafsson** [comment]
As having electronics engineer background, I would say this is incorrect schematic. Check how non-inverting optocoupler should be connected.

Don't use GND symbol if you don't intend to connect signal to ground.

ST driver output goes to the cathode of input of opto via current limiter resistor. Now there is also LED in series too, which should not be there. Not at least in the input side, could be  feasible in output if need to see if there is data traffic.

On the output connection should have ground connection, which is common with 3v3 system.

**Comment by MatsA** [comment]
@tkurki,  I have done a quick clearer ? schematic which I can add to the PR.
IMG_1163

**Comment by MatsA** [comment]
@KEGustafsson 

In both my blog, [https://pysselilivet.blogspot.com](https://pysselilivet.blogspot.com), and in contributions to Signal K I have tested and gone through the information very carefully.
 
The circuit is a working solution, not just an idea…… so in the connection settings in Signal K you should set the ”Invert Signal” to ”No”, optocoupler circuit, otherwise it will not work.

The LED is put at the input since it was used that way in the mentioned PCB. There is also a second thought that you should not add more load to the RPi 3,3 V. A LED on the +12 VDC side will indicate flow of data and there is no more load added.

Regarding the ground connection please check the schematic above.

**Comment by MatsA** [comment]
@KEGustafsson

I can just relate to what I have tested and got working so can’t change circuit to what Teppo linked to. The documentation is also referring to the mentioned PCB on Amazon.

Even if the voltage drop for the LEDs is 4 Volts and the current transfer ratio is down to 100 % there is a very big margin for a working output at the Rpi side.

The big thing is that the info in "Seatalk.md" is wrong and the latest circuit makes it more reliable. 

The picture in this PR was also the last picture in the old documentation.

@tkurki 
I can add the updated circuit or else we just drop the PR ?
In other words I will not "start a fight" to get things right according to my tests.

**Comment by KEGustafsson** [comment]
Don't take me wrong, but You have done great work updating ST documentation.

I'm a bit concervative in certain aspects. There are uncountable ways to do electronics circuits and make it work. Some are more common that others.

I think you could keep your way, but it could be usable to add also the other topology here too.

**Comment by MatsA** [comment]
@KEGustafsson OK, then we wait for @tkurki:s opinion.

We can just add the link to the forum.

It should also be said the it was me who tested this ST1 connection out, wrote the initial setup and @tkurki who implemented it in SK. Therefore I'm little more engaged than usual.... 😉

**Comment by tkurki** [comment]
My main concern was that the circuit diagram was super unclear (both existing and the new one). Your handdrawn one looks clear to me, so please can you tidy it up as a real diagram and we can proceed?

My €.02 about the led placement: 
1. is it an option and not part of the minimal setup? If an option you could indicate that in the diagram and in the text
2. led blinking on the input side does not necessarily mean that the output circuit is working and there is data there

**Comment by MatsA** [comment]
OK, I have changed the picture and would like to use 2 variants
A/ Using the mentioned PCB
B/ Using your own components.
Check pictures.

ST1_interface_PCB
ST1_interface_components

1/ Of course it's an option and it can be skipped for a minimal setup, but it was included in the PCB variant. I can skip it for variant B
2/ Thats correct, but on the other hand a blinking diod on the output just indicate that there is data not that it's correct...

Hopefully I can update the PR tomorrow...

**Comment by MatsA** [comment]
Will close this one and start a new one, seems to be easier ....
